### ISSUE TITLE
Remove `key` from `ManipulationMixin` and `AnalysisMixin`

### ISSUE THREAD

**Issue by t-reents** [aim2dat/aim2dat#68]
In the current version, all methods in `ManipulationMixin` and `AnalysisMixin` have a `key` argument. This is set to `None` in case the method is called on a `Structure` method. We only use it to select the desired structures in case the method is called on `StructureOperations`. 

I've been thinking about the following suggestion from time to time and finally write it down :D 

While the adjustment of the arguments doesn't really cost performance, personally, I'm not 100% happy with it (although I understand and remember our previous considerations). It feels a bit weird to have an additional argument in all the functions that is overridden for all the applications inside `Structure`.

https://github.com/aim2dat/aim2dat/blob/1952d240c726266c916a2f2383288a007ac79ff0/aim2dat/strct/mixin.py#L49-L51

My suggestion would be the following:
What about changing the syntax of `StructureOperations` from 

```
strct_ops = StructureOperations(structures)
strct_ops.any_method([1, 2, 4], *args)
```
to

```
strct_ops = StructureOperations(structures)
strct_ops[[1, 2, 4]].any_method(*args)
```

We already provide indexing and slicing which creates a new `SturctureOperations` object containing the subset of structures. Moreover, it neither complicates the code nor makes it longer. 
Since we don't store anything in the object but only in the `Structure` objects, returning slices is also not a problem. The result is returned and not stored anyway.

In this way, we could remove the key arguments and get an even more uniform interface between `Structure` and `StructureOperations`.

@hdsassnick @Scemp Comments are more than welcome, happy to discuss.

**Comment by hdsassnick** [comment]
Thanks @t-reents for this elegant solution, I am just not sure, how the implementation would look like and whether it can be parallelized?

**Comment by t-reents** [comment]
1. So the general implementation wouldn't change that much (at least in my dreams haha). We basically move the selection of which structures to consider for a certain operation from the function call to the indexing. Therefore, we could remove the following part and simply iterate over all the structures in the `StructureOperations`, as the selection was already done before

**Comment by t-reents** [comment]
Hi @hdsassnick 

The implementation would generally work. At the moment, one limiting factor is the appending to the current `StructureCollection` in the corresponding `StructureOperations` in case of the manipulation methods. I remember that I was actually voting against it at some point, but rethinking it now. How is your opinion on returning an object with a reference to the initial `StructureCollection` instead of copies/new objects when indexing `StructureOperations` objects. If we change this behaviour, I think I'll be able to fully implement the proposed change. 

The user experience will be similar to `pandas` if we decide to change the behaviour of indexing. In order to not modify the `StructureOperations` object, one would need to call something like `strct_op[[1,2,4]].copy()`. When `append_to_coll` is `False`, it doesn't matter anyway.

**Comment by hdsassnick** [comment]
Closed via #72.
